reporter/datadog_reporter.go

```diff
@@ -511,6 +526,22 @@ func (r *DatadogReporter) getPprofProfile() (profile *pprofile.Profile,
 			sample.Location = append(sample.Location, loc)
 		}
 
+		execPath, _ := r.execPathes.Get(trace.pid)
+
+		// Check if the last frame is a kernel frame.
+		if trace.frameTypes[len(trace.frameTypes)-1] == libpf.KernelFrame {
+			// If the last frame is a kernel frame, we need to add a dummy
+			// location with the kernel as the function name.
+			execPath = "kernel"
+		}
+
+		if execPath != "" {
+			loc := createPProfLocation(profile, 0)
+			m := createPprofFunctionEntry(funcMap, profile, "", execPath)
```

Comment:
minor, I think we could consider keeping only the executable as function name, then using the actual path as a file.
This would be mostly an aesthetic consideration, so we can easily fine tune this here or in the backend, or during a later stage.